app/src/main/assets/extensions/ads/ads.js

```diff
@@ -14,28 +14,39 @@ function collectLinks(urls) {
     }
 }
 
-function sendLinks() {
+function sendLinks(cookies) {
     let urls = [];
     collectLinks(urls);
 
     let message = {
      'url': document.location.href,
-     'urls': urls
+     'urls': urls,
+     'cookies': cookies
     };
     browser.runtime.sendNativeMessage("MozacBrowserAds", message);
 }
 
+function notify(message) {
+   sendLinks(message.cookies);
+}
+
+browser.runtime.onMessage.addListener(notify);
+
 const events = ["pageshow", "load", "unload"];
 var timeout;
 
 const eventLogger = event => {
   switch (event.type) {
     case "load":
-        timeout = setTimeout(sendLinks, ADLINK_CHECK_TIMEOUT_MS);
+        timeout = setTimeout(() => {
```

Comment:
Most of these changes are copied from https://github.com/mozilla-mobile/fenix/blob/07748f69b75144e8e247b742f66a756f999cb299/app/src/main/assets/extensions/cookies/cookies.js in order to retrieve the cookie information and send the cookies JSON message.